microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/faulttolerance/java/MicroProfileFaultToleranceDiagnosticsParticipant.java

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * Validates that the Fallback annotation's fallback method exists
+ *
+ */
+public class MicroProfileFaultToleranceDiagnosticsParticipant implements IJavaDiagnosticsParticipant {
+
+	@Override
+	public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		IJavaProject javaProject = context.getJavaProject();
+		return JDTTypeUtils.findType(javaProject, FALLBACK_ANNOTATION) != null;
+	}
+
+	@Override
+	public List<Diagnostic> collectDiagnostics(JavaDiagnosticsContext context, IProgressMonitor monitor)
+			throws CoreException {
+		ITypeRoot root = context.getTypeRoot();
+		IJavaElement[] elements = root.getChildren();
+		List<Diagnostic> diagnostics = new ArrayList<>();
+		collectDiagnostics(elements, diagnostics, context, monitor);
+		return diagnostics;
+	}
+
+	private static void collectDiagnostics(IJavaElement[] elements, List<Diagnostic> diagnostics,
+			JavaDiagnosticsContext context, IProgressMonitor monitor) {
+		for (IJavaElement element : elements) {
+			if (monitor.isCanceled()) {
+				return;
+			} else if (element.getElementType() == IJavaElement.TYPE) {
+				IType type = (IType) element;
+				validateClass(type, diagnostics, context, monitor);
+			}
+		}
+	}
+
+	private static void validateClass(IType type, List<Diagnostic> diagnostics, JavaDiagnosticsContext context,
+			IProgressMonitor monitor) {
+		CompilationUnit ast = context.getASTRoot(); //ASTResolving.createQuickFixAST(type.getCompilationUnit(), null);
+		ast.accept(new FaultToleranceAnnotationValidator(type, diagnostics, context));
+	}
+
+	private static class FaultToleranceAnnotationValidator extends ASTVisitor {
+
+		private IType type;
+		private Set<String> methodSet;
+		private List<Diagnostic> diagnostics;
+		private JavaDiagnosticsContext context;
+
+		private static Logger LOGGER = Logger.getLogger(FaultToleranceAnnotationValidator.class.getName());
+
+		public FaultToleranceAnnotationValidator(IType type, List<Diagnostic> diagnostics, JavaDiagnosticsContext context) {
+			super();
+			this.type = type;
+			this.methodSet = null;
+			this.diagnostics = diagnostics;
+			this.context = context;
+		}
+
+		@Override
+		public boolean visit(MethodDeclaration node) {
+			try {
+				validateMethod(node, type, diagnostics, context);
+			} catch (JavaModelException e) {
+				LOGGER.log(Level.WARNING, "An exception occured when attempting to validate the fallback method");
+			}
+			super.visit(node);
+			return true;
+		}
+
+		private void validateMethod(MethodDeclaration node, IType type, List<Diagnostic> diagnostics,
+				JavaDiagnosticsContext context) throws JavaModelException {
+			List modifiers = node.modifiers();
+			for (Object modifier : modifiers) {
+				if (modifier instanceof NormalAnnotation) {
+					NormalAnnotation annotation = (NormalAnnotation) modifier;
+					if (FALLBACK_ANNOTATION_ABBR.equals(annotation.getTypeName().getFullyQualifiedName())) {
```

Comment:
* Remove FALLBACK_ANNOTATION_ABBR 
 * create in AnnotationUtils:

```java
import org.eclipse.jdt.core.dom.Annotation;
....
public static boolean isMatchAnnotation(Annotation annotation, String annotationName) {
	return annotationName.endsWith(annotation.getTypeName().getFullyQualifiedName());
}

```* import static org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils.isMatchAnnotation;